cpp/src/strings/copy/repeat_strings.cu

```diff
@@ -0,0 +1,203 @@
+}
+
+namespace {
+/**
+ * @brief Generate a strings column in which each row is an empty or null string.
+ *
+ * The output strings column has the same bitmask as the input column.
+ */
+auto generate_empty_output(strings_column_view const& input,
+                           size_type strings_count,
+                           rmm::cuda_stream_view stream,
+                           rmm::mr::device_memory_resource* mr)
+{
+  auto chars_column = create_chars_child_column(strings_count, 0, stream, mr);
+
+  static_assert(std::is_same_v<offset_type, int32_t> && std::is_same_v<size_type, int32_t>);
+  auto offsets_column = make_numeric_column(
+    data_type{type_id::INT32}, strings_count + 1, mask_state::UNALLOCATED, stream, mr);
+  CUDA_TRY(cudaMemsetAsync(offsets_column->mutable_view().template begin<offset_type>(),
+                           0,
+                           (strings_count + 1) * sizeof(offset_type),
+                           stream.value()));
+
+  return make_strings_column(strings_count,
+                             std::move(offsets_column),
+                             std::move(chars_column),
+                             input.null_count(),
+                             input.null_count()
+                               ? cudf::detail::copy_bitmask(input.parent(), stream, mr)
+                               : rmm::device_buffer{},
+                             stream,
+                             mr);
+}
+
+/**
+ * @brief Functor to compute string sizes and repeat the input strings.
+ *
+ * This functor is called only when `repeat_times > 0`.
+ */
+struct compute_size_and_repeat_fn {
+  column_device_view const strings_dv;
+  size_type const repeat_times;
+  bool const has_nulls;
+
+  offset_type* d_offsets{nullptr};
+
+  // If d_chars == nullptr: only compute sizes of the output strings.
+  // If d_chars != nullptr: only repeat strings.
+  char* d_chars{nullptr};
+
+  // `idx` will be in the range of [0, repeat_times * strings_count).
+  __device__ void operator()(size_type const idx) const noexcept
+  {
+    auto const str_idx    = idx / repeat_times;
+    auto const repeat_idx = idx % repeat_times;
+    auto const is_valid   = !has_nulls || strings_dv.is_valid_nocheck(str_idx);
+
+    // Only `strings_count` threads will compute string sizes.
+    if (!d_chars && repeat_idx == 0) {
+      d_offsets[str_idx] =
+        is_valid ? repeat_times * strings_dv.element<string_view>(str_idx).size_bytes() : 0;
+    }
+
+    // Each input string will be copied by `repeat_times` threads into the output string.
+    if (d_chars && is_valid) {
+      auto const d_str    = strings_dv.element<string_view>(str_idx);
+      auto const str_size = d_str.size_bytes();
+      if (str_size > 0) {
+        auto const input_ptr  = d_str.data();
+        auto const output_ptr = d_chars + d_offsets[str_idx] + repeat_idx * str_size;
+        std::memcpy(output_ptr, input_ptr, str_size);
```

Comment:
@jrhemstad Do you have any recommendations on better device side memcpy?

Reply:
I vaguely remember that @davidwendt prefers `std::memcpy` over `thrust::copy(thrust::seq,...)`.....?

Reply:
After some more coffee I was thinking about `cuda::memcpy_async` as I thought it has better chances of being converted into vectorized stores